### PR TITLE
Allow alias usage for the domain and system owner

### PR DESCRIPTION
**Pull request by rocktavious**
Resolves https://github.com/OpsLevel/team-platform/issues/461

### Problem

Domain and System `owner` field cannot properly take a team alias.

### Solution

Uses the opslevel-go cache to lookup a team by the alias to find the ID to feed into the mutations.

### Checklist

- [ ] I have run this code, and it appears to resolve the stated issue.
- [ ] This PR does not reduce total test coverage
- [ ] This PR has no user interface changes or has already received approval from product management to change the interface.
- [ ] Make a [changie](https://github.com/OpsLevel/terraform-provider-opslevel/blob/main/CONTRIBUTING.md#changie-change-log-generation) entry that explains the customer facing outcome of this change

### CODE REVIEW

File: .changes/unreleased/Bugfix-20241224-105420.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+kind: Bugfix
+body: Actually allow opslevel_domain and opslevel_system resources to specify an alias
```
Comment (rocktavious) [inline .changes/unreleased/Bugfix-20241224-105420.yaml:2]:
```suggestion
body: Allow opslevel_domain and opslevel_system resources to specify an alias for their owner
```

File: opslevel/helpers.go [hunk 2]
```diff
@@ -15,3 +17,15 @@ func read[T any](ctx context.Context, d *diag.Diagnostics, state TerraformSource
...
 	return data
 }
+
+func GetTeamID(d *diag.Diagnostics, identifier string) *opslevel.ID {
+	if opslevel.IsID(identifier) {
+		return opslevel.NewID(identifier)
+	}
+	team, ok := opslevel.Cache.TryGetTeam(identifier)
+	if !ok {
+		d.AddError("opslevel error", fmt.Sprintf("failed to find team '%s'", identifier))
```
Comment (rocktavious) [inline opslevel/helpers.go:27]:
```suggestion
		d.AddError("opslevel error", fmt.Sprintf("failed to find team with alias '%s'", identifier))
```

File: opslevel/provider.go [hunk 3]
```diff
@@ -161,6 +161,7 @@ func (p *OpslevelProvider) Configure(ctx context.Context, req provider.Configure
 		tflog.Error(ctx, fmt.Sprintf("OpsLevel client validation error: %s", err))
 	}
 	tflog.Debug(ctx, "OpsLevel client is valid")
+	opslevel.Cache.CacheTeams(client)
```
Comment (rocktavious) [inline opslevel/provider.go:164]: @davidbloss do you know if this function is only called once for each `terraform plan` or `terraform apply` and not once per parallel resource action?

Comment (davidbloss) [reply opslevel/provider.go:164]: I haven't tested this out but I'm betting this is only called once per plan/apply

File: opslevel/resource_opslevel_domain.go [hunk 4]
```diff
@@ -88,7 +87,6 @@ func (r *DomainResource) Schema(ctx context.Context, req resource.SchemaRequest,
 			"owner": schema.StringAttribute{
 				Description: "The id of the team that owns the domain.",
```
Comment (davidbloss) [inline opslevel/resource_opslevel_domain.go:88]:
```suggestion
				Description: "The id or alias of the team that owns the domain.",
```

Comment (rocktavious) [reply opslevel/resource_opslevel_domain.go]: Fixed

File: opslevel/provider.go (hunk 3 above)
Comment (rocktavious) [reply opslevel/provider.go]: opted to go with a inline call to GetTeamWithAliases

**Review by davidbloss — APPROVED** [review]
LGTM 🎉

**Pull request merged**